cloud/blockstore/libs/storage/partition_nonrepl/part_mirror_resync_ut.cpp

```diff
@@ -1345,6 +1349,95 @@ Y_UNIT_TEST_SUITE(TMirrorPartitionResyncTest)
     {
         DoTestShouldTreatFreshDevicesProperly(false);
     }
+
+    Y_UNIT_TEST(ShouldHandleGetDeviceForRangeRequest)
+    {
+        TTestBasicRuntime runtime;
+        TTestEnv env(runtime);
+
+        TPartitionClient client(runtime, env.ActorId);
+
+        // Resync first range [0..1023]
+        env.CatchEvents(TEvNonreplPartitionPrivate::EvResyncNextRange);
+        env.StartResync();
+        env.ReleaseEvents();
+        env.ResyncController.WaitForResyncedRangeCount(1);
+
+        {   // Request to first device over an already resynchronized range
+            client.SendRequest(
+                env.ActorId,
+                std::make_unique<
+                    TEvNonreplPartitionPrivate::TEvGetDeviceForRangeRequest>(
+                    TBlockRange64::WithLength(0, 10)));
+            auto response = client.RecvResponse<
+                TEvNonreplPartitionPrivate::TEvGetDeviceForRangeResponse>();
+            UNIT_ASSERT_C(
+                SUCCEEDED(response->GetStatus()),
+                response->GetErrorReason());
+            UNIT_ASSERT_STRING_CONTAINS(
+                response->Device.GetDeviceUUID(),
```

Comment:
let's validate that we have correct `DeviceBlockRange` returned as well, here and in other tests